docs/design/proposals/healer-controller.md

```diff
@@ -0,0 +1,120 @@
+# Introduction
+There is [external health monitoring](https://github.com/kubernetes-csi/external-health-monitor/) which does mostly similar stuff, but actions are not in the scope of it, all it can do is log the health condition of the mounts.
+
+# Volume Healer Controller
+Before we begin it is noteworthy to know that the existing healer logic remains the same.
+
+**Now the obvious question we expect is "what do we achieve with the new controller?"**
+* Moving the k8s specific code out of ceph-csi, to make ceph-csi orchestrator independent.
+* Achieve a way to monitor the health of NBD volumes.
+* We currently do not have a way to act on failed/crashed NBD volumes with the existing healer logic. Once the controller detects that the volume is not healthy, it will have an opportunity to act on it.
```

Comment:
Do not focus on NBD, it is just one example. Why would NBD be different from other block devices or mounters?